src/connectdlg.cpp

```diff
@@ -731,6 +740,12 @@ void CConnectDlg::OnConnectClicked()
     done ( QDialog::Accepted );
 }
 
+void CConnectDlg::OnDeleteServerAddrClicked()
+{
+    pSettings->vstrIPAddress.erase ( pSettings->vstrIPAddress.begin() + cbxServerAddr->currentIndex() );
```

Comment:
Let's assume
`int iIdx = cbxServerAddr->currentIndex();` returns `MAX_NUM_SERVER_ADDR_ITEMS - 1` (which is the highest it can get to).
```
pSettings->vstrIPAddress[iIdx] = pSettings->vstrIPAddress[iIdx + 1];
```
This will then be indexing off the end of the list.
```
pSettings->vstrIPAddress[MAX_NUM_SERVER_ADDR_ITEMS - 1] = "";
```
This would then correctly set the last entry to empty.

Reply:
> Let's assume `int iIdx = cbxServerAddr->currentIndex();` returns `MAX_NUM_SERVER_ADDR_ITEMS - 1` (which is the highest it can get to).
> 
> ```
> pSettings->vstrIPAddress[iIdx] = pSettings->vstrIPAddress[iIdx + 1];
> ```
> 
> This will then be indexing off the end of the list.

No it won't, because the `for` condition will fail first. Note that the condition both specifies `MAX_NUM_SERVER_ADDR_ITEMS - 1` *and* uses `<` not `<=`. For that very reason.

> ```
> pSettings->vstrIPAddress[MAX_NUM_SERVER_ADDR_ITEMS - 1] = "";
> ```
> 
> This would then correctly set the last entry to empty.

Reply:
🤦 yes...

Anyway - the other issue I had was with `cbxServerAddr->currentIndex()` - a couple of times it seemed to point to the wrong item but I couldn't reproduce the problem reliably.  The string compare method I've pushed should
- find all multiple matches (just in case some have crept in)
- have no problems with entering a new value, _not_ hitting enter but hitting delete entry (I think this may have caused deleting the wrong entry...)
- have no problems with indexing the list

I've not yet checked but I'm hoping on write, all entries (including empty get written) but on read, only the non-empty ones get into current settings - so on next write, only the non-empty ones should be stored...  Ideally, on write, only non-empty ones would get written.